repository.go

```diff
@@ -534,6 +529,43 @@ func (r repository) update(cw contextWrapper, doc *Document, mutation Mutation,
 	return nil
 }
 
+func (r repository) applyMutates(cw contextWrapper, doc *Document, mutation Mutation, filter FilterQuery) error {
+	var (
+		baseQueries = []Querier{filter, mutation.Unscoped, mutation.Cascade}
+		queries     = baseQueries
```

Comment:
We have to reload mutation queriers (like version_lock += 1) when reloading a document, so we store the base slice